### DISCUSSION TITLE
System Status - No "Sensor Start/Age" display after xdrip update

### DISCUSSION THREAD

**Discussion by Teute0815** [NightscoutFoundation/xDrip#3744] [Bug reports]
I've been using
Libre3 Sensor -> juggluco (410 5.1.5) broadcasting through "Libre2 patched app" -> xdrip (dfcbe80-2024.09.17)
on a
Samsung S9+ Android 10
for quite some time.

The bug occured after automatic xdrip update through stable channel (dfcbe80-2024.09.17)
No other settings, versions or hardware have been changed.

After update, the Classic Status Page didn't display the sensor start date and the sensor age anymore. While being on a running sensor, I waited for that sensor to be replaced a few days later, to see if that would fix the "problem". It didn't. 

With my setting, xdrip doesn't display the sensor start date - and thus the sensor age (which is the quick information I'm missing most) anymore. I reinstalled the previous version again, to make sure it's an xdrip issue. It is - previous version instantly showed sensor start date and age again. It's not only a "**Libre2**" problem but also a "**Nightscout Follower**" source problem, which I tried in the meantime. (see images below) @Navid200 

Could that be fixed?

**source "Libre (patched App)"**
![2024-11-02 12 19 42](https://github.com/user-attachments/assets/b9ea3c24-4d3c-4cd1-ab43-55108e886a7a)

In the meantime I tried something new and got my juggluco readings via local webserver with xdrip as nightscout follower. That way I can access missed historical values, which is not possible via juggluco - libre patched.

When using xdrip as nightscout follower, xdrip got the exact same problem:

**source "Nightscout Follower"**
![2024-11-02 12 17 15](https://github.com/user-attachments/assets/6b275904-c16c-40ad-8026-7a28e5662ec7)

**Comment by Teute0815** [comment]
I wanted to change/swap the screenshots (smaller size), but "edit post" lets me access the headline of the topic only - I can't change the post (text or images) itself. Is that intended?

**Reply by Navid200** [reply]
No, you can edit the title, and you can edit each post individually.

**Comment by Navid200** [comment]
If you like to confirm that this is due to an update, you can install the version of xDrip you used to run over your updated version.  There is no need to uninstall.

If you can confirm that, it may help to figure out exactly what change in xDrip could have caused it.

**Reply by Teute0815** [reply]
Thanks for your reply. I did that. It's def an xdrip issue. And not only for "Libre2 patched" source but also for "Nightscout Follower" source.
See updated main post (Thanks for your your other tip ;-) images won't display smaller, though...)

**Reply by Navid200** [reply]
I only see references to one version of xDrip, which is the current stable release.

What I need are two xDrip releases to be consecutive Nightly releases one failing and one working.  That will tell me which Nightly change caused the problem.

Only then, can I see what changes we have made between those two releases and take it from there.

**Reply by Teute0815** [reply]
It's two different versions. The two latest **stable** releases - 2024.04.19 and 2024.09.17 which I got through the according update channel. I don't use/haven't used/tried any other Beta or Alpha versions
Two images with source "Libre 2"
Two images with source "Nightscout follower"
Both with each version. Old one (04.19) working - New one (09.17) not working

**Comment by Navid200** [comment]
@PhilTu Do you know what this is?

**Reply by PhilTU** [reply]
Not offhand. xDrip isn't connecting to the sensor but is receiving data from either Nightscout (data upload from Juggluco) or Juggluco emulating the patched app.

So any hypothetical change may be outside of xDrip Libre functionality.

**Comment by Teute0815** [comment]
I checked the nightly releases.

Bug occurs from 06.11 (working) to 06.25 (not working)
@Navid200 

![2024-11-02 12 16 28](https://github.com/user-attachments/assets/9b89c974-9296-4af1-89db-65f24aaeb376)

**Reply by Navid200** [reply]
That's odd,

None of the changes from 2024.06.11 to 2024.06.25 is related to Libre.

**Reply by Teute0815** [reply]
It's not solely "libre2 source" related. Same error occurs with other source "Nightscout Follower" (see images above). So, it's a more general issue of receiving status data - or maybe even more of displaying it, as the start date (but not the age, as that is not calculated/displayed there anyway) can still be seen on the "Libre2 status page" - so the value still arrives.

![2024-11-02 12 20 09](https://github.com/user-attachments/assets/ec45d8fc-18db-4360-86ed-320c3028ab3c)

**Reply by Navid200** [reply]
What exactly is the problem with Nightscout.
I upload my readings to Nightscout continuously.  I use the latest Nightly.  There is no problem with it.

Please tell me what the problem is with the follower.  I can try to recreate the problem.
Considering you say it is not related to Libre, I then should be able to see it as I don't use Libre.  I use Dexcom.

**Reply by Teute0815** [reply]
It's not a problem with Nightscout (we're talking about "two different Nightscout servers here"), but with the "Data source" setting and what is displayed in System Status. If you look at my images, it's two different source settings with the same display problem. Source "Libre (patched app)" as well as source "Nighscout Follower" don't show the Sensor Start Date and age anymore. 

I try to explain my setting a bit better/more in depth than I did in my original post:
I use **juggluco** to get my libre3 readings and send them to **xDrip**, which is my main/only app I "use" actively. xDrip then sends the data to Nightscout for documentation/reports purposes.

juggluco offers different ways to broadcast/send the values locally to xdrip
1. So far I used a broadcast which has to be fetched by xDrip with the source "**Libre (patched app)**"
2. juggloco also can run a local Web(Nightscout)server which xDrip can access through the source **"Nightscout Follower"**.

So, my Hardware/Software Setting is (either/or - both with same bug):

1. Libre3 Sensor -> juggluco -> **Patched Libre broadcast -> source "Libre patched"** -> xDrip-> Nightscout upload
2. Libre3 Sensor -> juggluco -> **Local Webserver -> source "Nightscout follower"** -> xDrip-> Nightscout upload

As I always missed "backfilling of values", which is not possible with juggluco - Libre patched - xDrip, I tried the local webserver, which can do that.

As far as I can tell, both alternatives work well and right and the same in both xDrip versions, except for the fact, that from 06.11 to 06.25 the displaying of the Sensor Start and Age got lost. The data itself still arrives as you can see on the last image "Libre Patched App status". 

Just out of interest: Could you post an image of your system status page? And maybe compare it to 06.11?

**Reply by Teute0815** [reply]
All in all it seems to be a common/any change somewhere which is not related to the source (the sensor not at all). Just like @PhilTU wrote.

**Reply by PhilTU** [reply]
> Please tell me what the problem is with the follower. 

The problem is the follower (xDrip) isn't displaying the sensor end time / age / start time.

**Reply by Navid200** [reply]
I removed sensor start time for G7 from the classic status page on June 25.  
I need to understand why this would impact your setup considering you don't use G7.

**Reply by Navid200** [reply]
@Teute0815 Click on `Create issue from discussion` in the right pane  so that we won't forget this.
I will need to fix this.
Thanks for reporting it.

**Reply by Teute0815** [reply]
@Navid200 
I have created the issue. Thanks for your reply. Why would you want to remove the sensor start/age for Dexcom from status page? Was that faulty? Otherwise a rollback of that specific setting would maybe be the best bet. For myself the classic status was always the quickest and easiest way to check sensor start and more importantly age, as the other according settings for libre (display on mainscreen, etc.) don't work anyway.

**Reply by Navid200** [reply]
G7 is a completely different mechanism compared to everything else we have had in xDrip so far.
What you see on the classic status page regardless of what sensor you use, is related to the xDrip session.  
The xDrip session is not always what counts.  For G7, the only session that really matters is the session on G7.
This is identical to G6.  So, one may ask why change this now and not leave it as it was with G6>

The answer is that because of the differences between G7 and G6, what was shown on the classic status page was very confusing.  We had no choice but to remove it.
If you are interested in more detail, you can see my explanation in the corresponding PR: https://github.com/NightscoutFoundation/xDrip/pull/3531  

So, no, we will not undo that.
But, I obviously have a mistake in it and need to fix it.
